E0232 needs to be updated to new format

From: src/test/compile-fail/E0232.rs

E0232 needs a span_label, updated title, and a note, updating it from:

```
error[E0232]: this attribute must have a value, eg `#[rustc_on_unimplemented = "foo"]`
  --> src/test/compile-fail/E0232.rs:13:1
   |
13 | #[rustc_on_unimplemented] //~ ERROR E0232
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
```

to:

```
error[E0232]: this attribute must have a value
  --> src/test/compile-fail/E0232.rs:13:1
   |
13 | #[rustc_on_unimplemented] //~ ERROR E0232
   | ^^^^^^^^^^^^^^^^^^^^^^^^^ attribute requires a value
   |
   = note: eg `#[rustc_on_unimplemented = "foo"]`
```

--- a/src/librustc_typeck/check/mod.rs
+++ b/src/librustc_typeck/check/mod.rs
@@ -903,9 +903,12 @@ fn check_on_unimplemented<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>,
                 }
             }
         } else {
-            span_err!(ccx.tcx.sess, attr.span, E0232,
-                                  "this attribute must have a value, \
-                                   eg `#[rustc_on_unimplemented = \"foo\"]`")
+            struct_span_err!(
+                ccx.tcx.sess, attr.span, E0232,
+                "this attribute must have a value")
+                .span_label(attr.span, &format!("attribute requires a value"))
+                .note(&format!("eg `#[rustc_on_unimplemented = \"foo\"]`"))
+                .emit();
         }
     }
 }

--- a/src/test/compile-fail/E0232.rs
+++ b/src/test/compile-fail/E0232.rs
@@ -10,7 +10,10 @@
 
 #![feature(on_unimplemented)]
 
-#[rustc_on_unimplemented] //~ ERROR E0232
+#[rustc_on_unimplemented]
+//~^ ERROR E0232
+//~| NOTE attribute requires a value
+//~| NOTE eg `#[rustc_on_unimplemented = "foo"]`
 trait Bar {}
 
 fn main() {

